### ISSUE TITLE
I believe that RuntimeExec class is creating threads that are not destroyed.

### ISSUE THREAD

**Issue by michaelgriffithus** [Alfresco/gytheio#2]
We are using Gytheio in an Alfresco project at Pearson Ed. We are seeing threads created that are not closed.  This is causing the native OS thread count to increase and never decrease. 

<img width="1434" alt="screenshot 2016-04-12 15 18 59" src="https://cloud.githubusercontent.com/assets/12091730/14472782/c639427c-00c2-11e6-855b-0a388be0cc4a.png">

<img width="1436" alt="screenshot 2016-04-12 15 27 30" src="https://cloud.githubusercontent.com/assets/12091730/14472836/16660758-00c3-11e6-89ad-d60c94a051e7.png">

**Comment by rgauss** [comment]
It's tough to give much feedback without the context of what it is you're trying to do, but in general the threads shouldn't hang around if the command line process is exiting normally.

You may have a problem with your stream handling and how the command you're calling interacts with it.

**Comment by michaelgriffithus** [comment]
We haven't modified the stream handling, we are using gytheio's RuntimeExec class just as its being used in the Alfresco media module.

**Comment by rgauss** [comment]
This is happening in an environment supported by the Media Management module?

Are you seeing this with ImageMagick or FFmpeg?

**Comment by michaelgriffithus** [comment]
Ray, we didn't use the Alfresco Media Module because it didn't cover our use case. Instead we rolled our own module that uses Gytheio on the back end.  Some mods were made to the code base because it was doing more than we needed and without any documentation we decided to simplify some of it. If you pm me at michael [dot] griffith [at] pearson.com I can send you the class where we are invoking RuntimeExec.
